web/src/ui_util.ts

```diff
@@ -7,15 +7,18 @@ import * as keydown_util from "./keydown_util";
 // dependencies other than jQuery.
 
 // https://stackoverflow.com/questions/4233265/contenteditable-set-caret-at-the-end-of-the-text-cross-browser
-export function place_caret_at_end(el: HTMLElement): void {
+export function place_caret_at_end(el: HTMLInputElement | HTMLElement): void {
```

Comment:
I think we do not need this change since `HTMLInputElement` inherits `HTMLElement`.